lib/game.rb

```diff
@@ -0,0 +1,128 @@
+class Game
+    attr_accessor :inputsX, :inputsO, :play_again, :valid_input, :active_player
+    def initialize(player1, player2)
+      @board = Board.new  
+      @playerX = Player.new(player1)
+      @playerO = Player.new(player2) 
+      @inputsX = []
+      @inputsO = []
+      @active_player = rand()*10 < 5 ? @playerO : @playerX
+      @play_again = true
+      @valid_input = false
+      start_game
+    end
+
+private
+     def start_game
```

Comment:
While it's good that you separated this method as a private method here are the problems I see with it:

1. This method is very complicated. Even as evidenced with the number of lines it has. This makes the Game class hard to understand and extend (if needed).
2. This method is mostly UI related than Class logic related. You want your Classes to be independent of the user-interface. As a good software design, following the tenets of OOP, it's imperative we make our code extendable. What I mean is this, suppose, for instance you were required to (by a client or a boss) to instead use a gem that will take the user input instead of writing your own? Then this means you will have to modify this Game class to accomodate that change. Yeah this seems a small matter right now in this case but imagine you have built a very big application.
3. This method shouldn't even be responsible for taking care of Player choices. It's the player class' responsibility to take the user choice.
4. You are calling your instance variables directly here. The reason we use `attr_reader` above the initialize statement is that, aside from making those attributes publicly readable, we also need to use them as such in our methods. For instance, instead of calling @valid_input  like this you could have simply written valid_input since it's in `attr_reader` already. But this isn't even the main problem here (1 to 3 are)